GFS_layer/GFS_driver.F90

```diff
@@ -266,9 +266,10 @@ subroutine GFS_initialize (Model, Statein, Stateout, Sfcprop,    &
 !-----------------------------------------------------------------------
 
     if (Model%do_cosp) then
-       do nb = 1, nblks
-          call cosp2_init (size(Grid(nb)%xlon,1), Model%levs)
-       enddo
+       !ONLY call with the first block, which will be equal to the blocksize.
+       ! this will (should?) allocate the right amount of data. Then calls to
+       ! COSP will have the right data set up.
+       call cosp2_init (size(Grid(1)%xlon,1), Model%levs)
```

Comment:
Please make sure this works when non-uniform blocks are created (e.g. [mod (nx, blocksize) != 0]

Reply:
My original code requires the block size to be the same as the layout size, making it much less efficient. Lucas's simple fix resolves this issue, although I haven't tested it in non-uniform blocks.